timefold-solver-python-core/src/main/python/_timefold_java_interop.py

```diff
@@ -257,6 +300,7 @@ def _add_to_compilation_queue(python_class: type | PythonSupplier) -> None:
 def _process_compilation_queue() -> None:
     global _compilation_queue
 
+    register_score_python_java_type_mappings()
```

Comment:
Why not adding this call to the `init()`function?

Reply:
Because then the score classes will not be in the generated classes directory if one is specified.

Reply:
I'm not following. In both cases,  `ensure_init()` is called, triggering `init()`. The `init()` function seems a better place for registering/initializing the map of types.

Reply:
Because `set_class_output_directory` may be called after `init`. This is where classes are compiled, which need to happen as late as possible; register_score_python_java_type_mapping` compiles the python score classes.